code/headers/uarm_gcode_generator.hpp

```diff
@@ -35,5 +110,310 @@ namespace r2d2::robot_arm {
+        void convert_coordinates_to_angle_of_joints_to_gcode(const vector3i_c &coordinate){
+            char x_string[11]; // max number of int digits (10) + '\0' = 11
+            char y_string[11];
+            char z_string[11];
+            this->int_to_string(coordinate.x, x_string);
+            this->int_to_string(coordinate.y, y_string);
+            this->int_to_string(coordinate.z, z_string);
+            this->append("#13 M2220 X");
+            this->append(x_string);
+            this->append(" Y");
+            this->append(y_string);
+            this->append(" Z");
+            this->append(z_string);
+            this->append("\n");
+        }
+
+        /**
+         * @brief convert angle of joints to coordinates to gcode
+         * Makes the convert angle of joints to coordinates gcode command for the uarm
+         * 
+         * @param angle angle of the joints: vector3("B joint 0","L joint 1","R joints 2") from 0~180 degree
+         */
+        void convert_angle_of_joints_to_coordinates_to_gcode(const vector3i_c &angle){
+            char b_string[11]; // max number of int digits (10) + '\0' = 11
+            char l_string[11];
+            char r_string[11];
+            this->int_to_string(angle.x, b_string);
+            this->int_to_string(angle.y, l_string);
+            this->int_to_string(angle.z, r_string);
+            this->append("#14 M2221 B");
+            this->append(b_string);
+            this->append(" L");
+            this->append(l_string);
+            this->append(" R");
+            this->append(r_string);
+            this->append("\n");
+        }
+
+        /**
+         * @brief check posibility of coordinates cartesian polar to gcode
+         * Makes the check posibility of coordinates cartesian polar gcode command for the uarm
+         * 
+         * @param coordinate cartesian integer coordinate(XYZ) 
+         * @param id 0 = cartesian , 1 = polar
+         */
+        void check_posibility_of_coordinates_cartesian_polar_to_gecode(const vector3i_c &coordinate, const uint8_t &id){
+            char x_string[11]; // max number of int digits (10) + '\0' = 11
+            char y_string[11];
+            char z_string[11];
+            char id_string[2];
+            this->int_to_string(coordinate.x, x_string);
+            this->int_to_string(coordinate.y, y_string);
+            this->int_to_string(coordinate.z, z_string);
+            this->int_to_string(id, id_string);
+            this->append("#15 M2222 X");
+            this->append(x_string);
+            this->append(" Y");
+            this->append(y_string);
+            this->append(" Z");
+            this->append(z_string);
+            this->append(" P");
+            this->append(id_string);
+            this->append("\n");
+        }
+
+    	/**
+         * @brief set current position head to reference position to gcode
+         * Makes the set current position head to reference position gcode command for the uarm
```

Comment:
Sentence does not flow very well.

Reply:
> It is a bit odd that a lot of the functions say 'This function wil get ...' however don't return that value

That's because I have not implemented a char for the debug method to return. It will be another feature/upgrade to get after this PR